riptide-compression/src/main/java/org/zalando/riptide/compression/Compression.java

```diff
@@ -0,0 +1,19 @@
+package org.zalando.riptide.compression;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apiguardian.api.API;
+import org.zalando.fauxpas.ThrowingUnaryOperator;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+@API(status = EXPERIMENTAL)
+@AllArgsConstructor(staticName = "of")
+@Getter
```

Comment:
Instead of `Getter` it could expose a decorate method directly

Reply:
```
@AllArgsConstructor(staticName = "of")
public final class Compression {
    @Getter
    private final String contentEncoding;
    private final ThrowingUnaryOperator<OutputStream, IOException> outputStreamDecorator;

    public OutputStream compress(final OutputStream stream) throws IOException {
        return outputStreamDecorator.tryApply(stream);
    }
}
```

Like this?

Working with `contentEncoding` may still be done by `RequestCompressionPlugin` (via the getter on `Compression`), or also the check for `arguments.getHeaders().containsKey(CONTENT_ENCODING)` should move IMO,

Reply:
Maybe something like this?

```
public interface Compression {

    public static final Compression GZIP = new DefaultCompression("gzip", GzipOutputStream::new);

    void update(HttpHeaders headers);
    OutputStream decorate(OutputStream stream) throws IOException;

}

Reply:
What do you think of:

> Working with contentEncoding may still be done by RequestCompressionPlugin (via the getter on Compression), or also the check for arguments.getHeaders().containsKey(CONTENT_ENCODING) should move IMO,

Is related to your comment in https://github.com/zalando/riptide/pull/822#discussion_r347265250